dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -293,6 +304,32 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context, CancellationToken
         return this;
     }
 
+    private async Task<SKContext> CallPostExecutionHook(SKContext context)
```

Comment:
Curious if we can just depend on the fact that a Plan is an ISKFunction and that things will inherit appropriately.

Reply:
Nope, I started that way but it breaks down in those scenarios:

1. When Plans are not functions/steps. (Steps within the plan will not trigger the the hook)
2. When new steps are added to the plan. (Those will not have the hooks, so the caller would need to manage that)
3. When hooks are cancelled at the plan level I would have to manage the multiple references in the steps level (for each function individually)
4. Some more :)

Reply:
@shawncal Agree, I think this decomposition of the TemplateEngine in a TemplatePlanner is something we need to address shortly. Using the Template as Plan, is another think we would need to think if by design the Hooks defined in the plan should be triggered for `TemplateGeneration` functions. But this could be something we can address at the Hooking set definition for `IPlan` interfaces.